### PR TITLE
Ports sidecar e2e harness to go

### PR DESCRIPTION
**Pull request by bowei**

### CODE REVIEW

File: cmd/sidecar-e2e/main.go [hunk 1]
```diff
@@ -136,27 +170,157 @@ func startDumpToBuf(lock *sync.Mutex, out *string, cmd *exec.Cmd) {
...
 
 func (h *harness) runTests() {
+	dir, err := filepath.Abs(h.tmpDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+	args := []string{"run", "--rm=true", "-v", fmt.Sprintf("%v:/t", dir), h.image}
+	log.Printf("docker %v", args)
+
+	cmd := exec.Command("docker", args...)
```
Comment (MrHohn) [inline cmd/sidecar-e2e/main.go]: Could we use h.docker() here? Seems duplicated.

File: cmd/sidecar-e2e/main.go [hunk 2]
```diff
@@ -23,46 +23,80 @@ import (
...
+}{
+	"harness",
+	false,
+	".",
+	"Dockerfile.e2e",
+	"/usr/sbin/dnsmasq",
+	"/sidecar",
+	"/usr/bin/dig",
+
+	"/t",
```
Comment (MrHohn) [inline cmd/sidecar-e2e/main.go]: May be a better name for this dir?

File: cmd/sidecar-e2e/main.go [hunk 3]
```diff
@@ -23,46 +23,80 @@ import (
...
+	flag.StringVar(&opts.digBinary, "digBinary", opts.digBinary,
+		"location of dig")
+
+	flag.StringVar(&opts.outputDir, "outputDir", opts.outputDir,
+		"location of output dir inside container")
 
 	flag.Parse()
 }
 
+func logWithPrefix(prefix string, str string) {
```
Comment (MrHohn) [inline cmd/sidecar-e2e/main.go]: Seems like it logs some empty outputs:
```
2016/12/29 21:36:32 docker | 2016/12/29 21:36:25 Running as test
2016/12/29 21:36:32 docker | 2016/12/29 21:36:25 Starting dnsmasq [-q -k -a 127.0.0.1 -p 10053 -c 1337 -8 - -A /ok.local/1.2.3.4 -A /nxdomain.local/]
...
```

File: cmd/sidecar-e2e/main.go (hunk 1 above)
Comment (bowei) [inline cmd/sidecar-e2e/main.go]: ok

File: cmd/sidecar-e2e/main.go (hunk 2 above)
Comment (bowei) [inline cmd/sidecar-e2e/main.go]: done

File: cmd/sidecar-e2e/main.go (hunk 1 above)
Comment (bowei) [inline cmd/sidecar-e2e/main.go]: done

File: cmd/sidecar-e2e/main.go (hunk 3 above)
Comment (bowei) [inline cmd/sidecar-e2e/main.go]: just a newline in the output stream

**Pull request merged**